src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py

```diff
@@ -68,18 +68,10 @@
     type: command
     short-summary: Create a lock.
     long-summary: 'Locks can exist at three different scopes: subscription, resource group and resource.'
-    examples:
-        - name: Create a read-only subscription level lock.
-          text: >
-            az lock create --name lockName --resource-group group --lock-type ReadOnly
```

Comment:
This should *definitely* not be removed. The low-level lock commands are extremely confusing. It is not obvious that using only the minimum parameters creates a subscription level lock.

Reply:
Also, if anything additional examples should be added to show creating resource group and resource level locks. Inadvertently missing a parameter results in a completely different and unexpected type of lock being created.

Reply:
> This should definitely not be removed. The low-level lock commands are extremely confusing. It is not obvious that using only the minimum parameters creates a subscription level lock.

This is a general documentation problem you're describing, not an examples-related problem. If it is not clear to users which levels various `lock` commands operate at, an example is not going to clarify that.

Reply:
It's an examples problem if the user looks to the examples first for guidance. There should be examples for this specific command for subscription, resource group and resource level locks.  There are more specific commands elsewhere that eliminate this confusion, but for this specific (legacy) command, removing the examples is not helpful.

CanNotDelete means that the object or resource group cannot be deleted. You can delete the lock (unless I suppose you lock the lock with another lock... o_O)

--resource can be a name, in which case you need --resource-type (and possibly other things) or it could be an ID, in which case that's all you need.